src/main/java/com/direwolf20/buildinggadgets/common/tools/gadget/building/IBuildingMode.java

```diff
@@ -0,0 +1,15 @@
+package com.direwolf20.buildinggadgets.common.tools.gadget.building;
+
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+
+import java.util.List;
+
+public interface IBuildingMode {
+
```

Comment:
To summarize my notes for additional Methods:
 
- an getter for this buildingModes id
- a translate Method.